### DISCUSSION TITLE
editOnClick for collection datasource

### DISCUSSION THREAD

**Discussion by thedangler** [Power-Components/livewire-powergrid#723] [Q&A]
I want to have edit on click for entering missing email addresses that come from a 3rd party data source.
My plan was to override onUpdateEditable to update the value with an API call to the 3rd party API.

But when I  try
```
Column::make('Email', 'Email')
                ->searchable()
                ->sortable()->editOnClick(true)
```

I get en error "argument #3 ($row) must be of type Illuminate\Database\Eloquent\Model, stdClass given"

Anyone figure this out?

**Comment by luanfreitasdev** [comment]
Actually for now I don't use editOnClick using collection as data source. If you want to work around the problem and send a PR I can help with any questions 😃

**Reply by thedangler** [reply]
I mean, I'd love to help.
Can you point me in the correct direction to get this started?
I published the component files.

**Reply by BeEasyCoding** [reply]
Is there a technical reason why editOnClick() couldn't be used on Collection datasources?
